client/src/main/java/io/spine/client/VersionColumn.java

```diff
@@ -24,37 +24,52 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-package io.spine.server.entity.storage;
+package io.spine.client;
 
 import io.spine.core.Version;
-import io.spine.query.ColumnName;
-import io.spine.query.CustomColumn;
-import io.spine.server.entity.Entity;
 
 /**
- * A column of an entity to store the {@linkplain Entity#version() entity version}.
+ * A column of an entity holding the entity version.
+ *
+ * <p>Extends the client-side query language to provide an ability to query entities
+ * by their versions.
+ *
+ * <pre>
+ *  Query query =
+ *        ReportView.query()
+ *                    // ...
+ *                  .where(Version.is(), previousVersion)
+ *                  .build(...);
+ *  ... = execute(query);
+ * </pre>
+ *
+ * <p>This type is a singleton.
  */
-final class VersionColumn extends CustomColumn<Entity<?, ?>, Version> {
+public final class VersionColumn extends EntityLifecycleColumn<Version> {
 
-    @SuppressWarnings("DuplicateStringLiteralInspection")   // Used in a different context.
-    private static final ColumnName VERSION = ColumnName.of("version");
-
-    VersionColumn() {
-        super();
-    }
+    private static final VersionColumn instance = new VersionColumn();
 
-    @Override
-    public ColumnName name() {
-        return VERSION;
+    /**
+     * Creates an instance of this column.
+     */
+    @SuppressWarnings("DuplicateStringLiteralInspection")   /* Used in a different context. */
+    private VersionColumn() {
+        super("version");
     }
 
-    @Override
-    public Class<Version> type() {
-        return Version.class;
+    /**
+     * Returns a singleton instance of this type.
+     */
+    public static VersionColumn instance() {
+        return instance;
     }
 
-    @Override
-    public Version valueIn(Entity<?, ?> entity) {
-        return entity.version();
+    /**
+     * Returns a singleton instance of this type.
+     *
+     * <p>Serves as a more DSL-friendly alternative to {@link #instance() instance()}.
+     */
+    public static VersionColumn is() {
+        return instance;
```

Comment:
So we should use `instance()` here over the direct access just as in the other columns.